### PR TITLE
Report StopIteration if connection is closed

### PR DESCRIPTION
**Pull request by crazyxy**
**What this PR does / why we need it**: Fix segment fault.

**Which issue this PR fixes** : fixes https://github.com/istio/istio/issues/14707

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by crazyxy** [comment]
> This seems a envoy bug, can you file an issue in envoy and add a TODO saying this is a workaround for that?

I also made the change for upstream Envoy. I am writing test for the change and will send the pr later.

File: src/envoy/tcp/mixer/filter.cc [hunk 1]
```diff
@@ -167,6 +169,7 @@ void Filter::completeCheck(const CheckResponseInfo &info) {
 void Filter::onEvent(Network::ConnectionEvent event) {
   if (filter_callbacks_->upstreamHost()) {
     ENVOY_LOG(debug, "Called tcp filter onEvent: {} upstream {}",
+              filter_callbacks_->connection(),
```
Comment (duderino) [inline src/envoy/tcp/mixer/filter.cc]: I think you need to add a third {} to the debug log

Comment (crazyxy) [reply src/envoy/tcp/mixer/filter.cc]: Good catch. I think `ENVOY_CONN_LOG` should be used here.

**Comment by bianpengyuan** [comment]
@crazyxy just FYI I am investigating prow build failure.

**Pull request merged**